### PR TITLE
#175636852 Add User Role-Setting for Super Administrator to Assign Roles to Users Using Their Emails

### PR DESCRIPTION
**Pull request by Thiernope**
#### What does this PR do?
Add User Role-Setting for Super Administrator to Assign Roles to Users Using Their Emails
#### Description of Task to be completed?

- create userRole models 
- add column of roles in the User model
- create superAdmin role
- Add functionalities to create, read, update and delete users roles using their emails

#### How should this be manually tested?

- run `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- run `cd directory tech-strikers-bn-backend`
- run `git checkout ft-notify-on-edit-175636861`

#### Any background context you want to provide?
no info so far
#### What are the relevant pivotal tracker stories?
[#175636852](https://www.pivotaltracker.com/n/projects/2475704/stories/175636852)
#### Screenshots (if appropriate)
No so far
#### Questions:
No so far

### CODE REVIEW

**Review by Duncanwar — COMMENTED** [review]
I am no seeing any file changed

File: src/database/models/user.js [hunk 1]
```diff
@@ -6,10 +6,17 @@ export default (sequelize, DataTypes) => {
       email: DataTypes.STRING,
       username: DataTypes.STRING,
       password: DataTypes.STRING,
-      role: DataTypes.STRING,
+      roleId: DataTypes.STRING,
```
Comment (hervebutera) [inline src/database/models/user.js:9]: roleId should be an integer

File: src/database/migrations/20201118122004-create-user.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ module.exports = {
       email: {
         type: Sequelize.STRING,
       },
-      role: {
+      roleId: {
         type: Sequelize.STRING,
         defaultValue: "user",
```
Comment (hervebutera) [inline src/database/migrations/20201118122004-create-user.js:27]: roleId type should be integer and the default value should no longer be 'user'

File: src/database/superAdminRole.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const createAdminRole = async () =>{
+    const role = {
+        name: "SUPER_ADMIN",
+        description: "This is the super Admin or IT stuff responsible for assigning roles to other users",
+        createdAt : new Date(),
+        updatedAt: new Date()
+    }
+    await userRoles.create(role);
+    console.log("Super Admin role has been created")
+}
```
Comment (hervebutera) [inline src/database/superAdminRole.js:13]: Console logging not advisable for production

File: src/database/superAdmin.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        email: "admin@gmail.com",
+        password: hashedPassword,
+        isVerified: true,
+        roleId: role.id,
+        createdAt: new Date(),
+        updatedAt: new Date()
+    };
+
+    await Users.create(superAdmin);
+    console.log("super admin has been created")
```
Comment (hervebutera) [inline src/database/superAdmin.js:33]: Console logging not advisable for production

**Review by hervebutera — CHANGES_REQUESTED** [review]
Left some comments and check for similar issues that reviewers may not have seen.

**Pull request closed without merging**